packages/apps/sendgrid/src/functios-to-sendgrid/abandoned-carts.ts

```diff
@@ -0,0 +1,74 @@
+import logger from 'firebase-functions/lib/logger';
+import api from '@cloudcommerce/api';
+// eslint-disable-next-line import/no-unresolved
+import { getFirestore } from 'firebase-admin/firestore';
+import { sgSendMail } from './utils';
+import parseCartToSend from './parse-data-to-send';
+
+export default async () => {
+  try {
+    logger.log('# Check abandoned carts');
+    const [application] = (await api.get(
+      'applications?app_id=129856&fields=hidden_data,data',
+    )).data.result;
+
+    const appData = {
+      ...application.data,
+      ...application.hidden_data,
+    };
+
+    let filterCartsDate = '';
+    const limitDate = new Date();
+    limitDate.setHours(limitDate.getHours() - (appData.abandoned_cart_delay || 3));
+    const lastNotificationSent = (await getFirestore().doc('sendGrid/lastCartSent').get()).data();
+    if (lastNotificationSent && lastNotificationSent.time) {
+      const lastNotifiedCart = new Date(lastNotificationSent.time);
+      filterCartsDate += `&create_at>=${lastNotifiedCart.toISOString()}`;
+    }
+
+    filterCartsDate += `&created_at<=${limitDate.toISOString()}`;
```

Comment:
Tem que haver uma data mínima também (`&created_at>=...`), pense principalmente se o app for instalado só depois já com a loja funcionando a um bom tempo, pode ter uma fila infinita aqui...
Acho que podemos usar 8x `(appData.abandoned_cart_delay || 3)` (daria 1 dia por padrão), com limite máximo para 1 mês.